cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -535,6 +536,45 @@ pub(super) fn gcroots_dir(flox: &Flox, owner: &EnvironmentOwner) -> PathBuf {
     flox.cache_dir.join(GCROOTS_DIR_NAME).join(owner.as_str())
 }
 
+/// Detects whether the the provided error is regarding an unsupported package for the current system
+pub fn is_unsupported_pkg_for_system_error(err: &CoreEnvironmentError) -> bool {
+    matches!(
+        err,
+        CoreEnvironmentError::LockedManifest(LockedManifestError::BuildEnv(
+            CallPkgDbError::PkgDbError(PkgDbError {
+                exit_code: error_codes::PACKAGE_EVAL_INCOMPATIBLE_SYSTEM,
+                ..
+            })
+        ),)
+    )
+}
+
+/// Converts the provided error into a variant that will be wrapped in an error that provides the doc link
+pub fn apply_doc_link_for_unsupported_packages(err: CoreEnvironmentError) -> CoreEnvironmentError {
+    if let CoreEnvironmentError::LockedManifest(LockedManifestError::BuildEnv(
+        CallPkgDbError::PkgDbError(PkgDbError {
+            exit_code: error_codes::PACKAGE_EVAL_INCOMPATIBLE_SYSTEM,
+            category_message,
+            context_message,
+        }),
+    )) = err
+    {
+        debug!("incompatible package, directing user to docs");
+        CoreEnvironmentError::LockedManifest(LockedManifestError::UnsupportedPackageWithDocLink(
+            CallPkgDbError::PkgDbError(PkgDbError {
+                exit_code: error_codes::PACKAGE_EVAL_INCOMPATIBLE_SYSTEM,
+                category_message,
+                context_message,
+            }),
+        ))
+    } else {
+        warn!("given an unexpected error variant for wrapping");
+        // This branch is only here to guard against misuse i.e. it should be unreachable when used
+        // correctly. Mistakenly not directing users to the docs is a better failure mode than a panic.
+        err
+    }
+}
```

Comment:
suggestion: move to flox crate
suggestion: maybe-wrap error with additional context:
```rust
apply_doc_link_for_unsupported_packages(err: CoreEnvironmentError) -> Result<CoreEnvironmentError, anyhow::Error> {
	match err {
		CoreEnvironmentError::LockedManifest(LockedManifestError::BuildEnv(
            CallPkgDbError::PkgDbError(PkgDbError {
                exit_code: error_codes::PACKAGE_EVAL_INCOMPATIBLE_SYSTEM,
                context_message,
            })
        ) => {
			
			// or use the format_* function
			let error_description = context_message
				.as_deref()
				.unwrap_or("This package is not available on this system");
			            
			Err(anyhow!(formatdoc!{"
				{error_description}
				
				<hint>
				<link>
				",
			}))
		}

	}
}
```

This can then be dropped into install and edit as is (though the semantics with Ok/Err are a bit weird admittedly, we can change that):

```
let other_error = apply_doc_link_for_unsupported_packages(err)?;
```

I think that a more granular way that can be shared between multiple commands without duplicating logic.

Alternatively we can similarly extract just the message in the pull impl which does not include the link.

Either way, it would be even better if we hadn't have to special case the formatting and just used a common format with the link in all cases,

Reply:
I think I may also prefer having an `anyhow!` error to a deeply nested type of ours - since once we catch it we know we want to print it, that would let us drop some of the re-matching in errors.rs. But I think we're starting to spend a bit too long in this and if it works, it's okay